README.md

```diff
@@ -1,2 +1,110 @@
-# terraform-google-log-export
-A Cloud Foundation Toolkit Module: Opinionated Google Cloud Platform project creation and configuration with Shared VPC, IAM, APIs, etc.
+# Terraform Logsink Module
+
+## Requirements
+### Terraform plugins
+- [Terraform](https://www.terraform.io/downloads.html) 0.10.x
+- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) plugin v1.8.0
+
+### Configure a Service Account
+In order to execute this module you must have a Service Account with the following:
+
+#### Roles
+The service account with the following roles:
+- roles/logging.configWriter on the logsink project (to create logsinks)
+- roles/serviceAccount.admin on the logsink project (to create a service account for the log sinks)
+- roles/iam.admin on the logsink project (to grant permissions to the service account)
+- roles/serviceusage.admin on both the logsink and the destination project (to enable API for destinations)
+- roles/storage.editor on the destination project (if logsink destination is a Cloud Storage bucket)
+- roles/pubsub.editor on the destination project (if logsink destination is a Cloud Pub/Sub topic)
+- roles/bigquery.dataEditor on the destination project (if logsink destination is a BigQuery table)
+
+
+### Enable API's
+In order to operate with the Service Account you must activate the following API's on the base project where the Service Account was created:
+
+- Cloud Resource Manager API - cloudresourcemanager.googleapis.com
+- Cloud Billing API - cloudbilling.googleapis.com
+- Identity and Access Management API - iam.googleapis.com
+- Service Usage API - serviceusage.googleapis.com
+
+## Install
+
+### Terraform
+Be sure you have the correct Terraform version (0.10.x), you can choose the binary here:
+- https://releases.hashicorp.com/terraform/
+
+## Usage
+You can go to the `examples` folder to see all the use cases, however the usage of the module could be like this in your own `main.tf` file:
+
+```
+    locals {
+      credentials_file_path = "<PATH TO THE SERVICE ACCOUNT JSON FILE>"

```

Comment:
Since we don't require the credentials_file for the module itself we should drop this requirement throughout (including in examples) so people can use Application Default credentials instead of they prefer.

Usage section in README can skip directly to module invocation.

Reply:
Right, will remove, but right now it's not immedialely clear to customers that they need to init the `google` provider before using our modules (IMO we're missing the bigger picture README).

Reply:
@ocervell We can include a link to the [Google Provider](https://www.terraform.io/docs/providers/google/index.html) in the README to clarify, but since there are multiple ways of configuring the Terraform provider (including environment variables) I don't want to be overly prescriptive here.

Reply:
sounds good, will strip it off

Reply:
FWIW, TF will bail very quickly with a very sane/clear message politely asking the user to run `terraform init`.